dashboard/app/models/levels/pythonlab.rb

```diff
@@ -99,4 +110,45 @@ def get_validations
       nil
     end
   end
+
+  def summarize_for_lab2_properties(script, script_level = nil, current_user = nil)
+    level_properties = super
+    level_properties[:serializedMaze] = get_serialized_maze
+    level_properties
+  end
+
+  def get_serialized_maze
+    serialized_maze || project_template_level&.try(:serialized_maze)
+  end
+
+  def clean_up_mini_app_settings
+    if mini_app != 'neighborhood'
```

Comment:
Nice!

Reply:
Nit: could add a test for this new feature if not too much trouble

Reply:
added!